libcaf_core/src/scheduled_actor.cpp

```diff
@@ -96,22 +96,24 @@ void scheduled_actor::default_exit_handler(scheduled_actor* ptr, exit_msg& x) {
 }
 
 #ifdef CAF_ENABLE_EXCEPTIONS
-error scheduled_actor::default_exception_handler(pointer ptr,
+error scheduled_actor::default_exception_handler(local_actor* ptr,
                                                  std::exception_ptr& x) {
   CAF_ASSERT(x != nullptr);
   try {
     std::rethrow_exception(x);
-  } catch (const std::exception& e) {
+  } catch (std::exception& e) {
```

Comment:
Why loose the `const` here?

Reply:
It just makes little sense to capture exceptions by const reference. Capturing by reference is the idiomatic way and there's always going to be a single reference to the exception anyway.

Reply:
k